src/dstack/_internal/core/backends/gcp/compute.py

```diff
@@ -116,7 +136,68 @@ def create_instance(
+        if "tpu" in instance_offer.instance.resources.gpus[0].name:
+            tpu_node = gcp_resources.create_tpu_node_struct(
+                instance_name=instance_offer.instance.name,
+                startup_script=startup_script,
+                authorized_keys=authorized_keys,
+                spot=instance_offer.instance.resources.spot,
+                project_id=self.config.project_id,
+                zone=instance_offer.region,
+                instance_id=instance_id,
+                disk_size=disk_size,
+                is_pod=_is_pod(instance_offer.instance.name),
+            )
 
+            create_node_request = tpu_v2.CreateNodeRequest(
+                parent=f"projects/{self.config.project_id}/locations/{instance_offer.region}",
+                node_id=instance_id,
+                node=tpu_node,
+            )
+            try:
+                operation = self.tpu_client.create_node(request=create_node_request)
+                gcp_resources.wait_for_operation(operation, verbose_name="tpu instance creation")
+            except (
+                google.api_core.exceptions.ServiceUnavailable,
+                google.api_core.exceptions.NotFound,
+                google.api_core.exceptions.ResourceExhausted,
+            ):
+                # There is no clear instruction on how to configure auto delete of persistent disk on
+                # termination of associated TPU VM. GCP does not provide documentation on
+                # how to get the vm instance name of the TPU VM. It only provides node name.
+                if gcp_resources.is_disk_present(
+                    project_id=self.config.project_id,
+                    zone=instance_offer.region,
+                    disk_name=f"{self.config.project_id}-{instance_id}",
+                ):
+                    gcp_resources.delete_persistent_disk(
+                        project_id=self.config.project_id,
+                        zone=instance_offer.region,
+                        disk_name=f"{self.config.project_id}-{instance_id}",
+                    )
+                raise NoCapacityError()
+            node_request = tpu_v2.GetNodeRequest(
+                name=f"projects/dstack/locations/{instance_offer.region}/nodes/{instance_id}",
+            )
+            instance = self.tpu_client.get_node(request=node_request)
+            return JobProvisioningData(
+                backend=instance_offer.backend,
+                instance_type=instance_offer.instance,
+                instance_id=instance_id,
+                hostname=instance.network_endpoints[0].access_config.external_ip,
+                internal_ip=None,
+                region=instance_offer.region,
+                price=instance_offer.price,
+                ssh_port=22,
+                username="ubuntu",
+                ssh_proxy=None,
+                dockerized=True,
+                backend_data=None,
+            )
         labels = {
```

Comment:
TPUs should be labeled as well. I believe they support labels, right?

Reply:
Yes I found out TPUs does support labels. Will fix it.